pkg/ansible/generate.go

```diff
@@ -140,79 +144,243 @@ openshift_aws_create_s3: false
+`
+	DefaultInventory = `
+[OSEv3:children]
+masters
+nodes
+etcd
+
+[OSEv3:vars]
+ansible_become=true
+
+[masters]
+
+[etcd]
+
+[nodes]
+`
 )
 
-type VarsGenerator struct {
-	Vars map[string]string
+type clusterParams struct {
+	Name       string
+	Region     string
+	SSHKeyName string
+	DefaultAMI string
 }
 
-func NewVarsGenerator(cluster *coapi.Cluster) *VarsGenerator {
-
-	// Any vars that change from cluster to cluster need to be defined here:
-	inv := &VarsGenerator{
-		Vars: map[string]string{
-			cloudProviderKindVar: "aws",
-			awsRegionVar:         "us-east-1",
-			awsClusterIDVar:      cluster.Name,
-			elbBasenameVar:       cluster.Name,
-			awsVPCNameVar:        cluster.Name,
-			awsAMIVar:            fmt.Sprintf("%s-ami-base", cluster.Name),
-		},
-	}
-	return inv
+type machineSetParams struct {
+	Name         string
+	Size         int
+	AMIName      string
+	InstanceType string
 }
 
-// GenerateVars returns a string suitable for use as a vars file when calling openshift-ansible pods.
-func (ig *VarsGenerator) GenerateVars() (string, error) {
+func GenerateClusterVars(cluster *coapi.Cluster) (string, error) {
+
+	// Currently only AWS is supported. If we don't have an AWS cluster spec, return an error
+	if cluster.Spec.Hardware.AWS == nil {
+		return "", fmt.Errorf("no AWS spec found in the cluster, only AWS is currently supported")
+	}
+
 	// Change template delimiters to avoid conflict with {{ }} use in ansible vars:
-	t, err := template.New("varsfilecontents").Delims("[[", "]]").Parse(varsTemplate)
+	t, err := template.New("clustervars").Delims("[[", "]]").Parse(clusterVarsTemplate)
 	if err != nil {
 		return "", err
 	}
 
+	params := clusterParams{
+		Name:       cluster.Name,
+		Region:     cluster.Spec.Hardware.AWS.Region,
+		SSHKeyName: cluster.Spec.Hardware.AWS.KeyPairName,
+	}
+
+	if cluster.Spec.DefaultHardwareSpec != nil && cluster.Spec.DefaultHardwareSpec.AWS != nil {
+		params.DefaultAMI = cluster.Spec.DefaultHardwareSpec.AWS.AMIName
+	}
+
 	var buf bytes.Buffer
-	err = t.Execute(&buf, ig.Vars)
+	err = t.Execute(&buf, params)
 	if err != nil {
 		return "", err
 	}
 	return buf.String(), nil
 }
+
+func GenerateMachineSetVars(cluster *coapi.Cluster, machineSet *coapi.MachineSet) (string, error) {
+	commonVars, err := GenerateClusterVars(cluster)
+	if err != nil {
+		return "", err
+	}
+
+	if machineSet.Spec.Hardware.AWS == nil {
+		return "", fmt.Errorf("no AWS spec found on machine set, only AWS is currently supported")
+	}
+
+	var buf bytes.Buffer
+	buf.WriteString(commonVars)
+
+	var varsTemplate string
+	switch {
+	case machineSet.Spec.NodeType == coapi.NodeTypeMaster:
+		varsTemplate = masterVarsTemplate
+	case machineSet.Spec.Infra:
+		varsTemplate = infraVarsTemplate
+	default:
+		varsTemplate = computeVarsTemplate
+	}
+
+	t, err := template.New("machinesetvars").Delims("[[", "]]").Parse(varsTemplate)
+	if err != nil {
+		return "", err
+	}
+
+	params := &machineSetParams{
+		Name:         machineSet.Name,
```

Comment:
Do we want this to be the full name of the machine set, with the cluster name and the generated suffix? Or do we want this to be the simpler name from `ClusterMachineSet`? Or does it not matter because it is only used and seen by the ansible playbook?

Reply:
It does go beyond just the playbook because it goes on the labels on the AWS asg. I did think about using the shorter name from the ClusterMachineSet. However, it is conceivable that at some point (during an upgrade or whatever), you could end up with more than one ASG with the same name/label, which would be confusing.